### ISSUE TITLE
Support for STM32F446 and STM32F469

### ISSUE THREAD

**Issue by davids5** [libopencm3/libopencm3#675]
There are a few specific changes to support the STM32F446 and STM32F469

1) The RCC start up code needs to not clear the PLLR bits. Bits 30:28 PLLR[2:0]: Main PLL division factor for DSI clock
2) In the USB device init OTG_FS_GCCFG ST changed the definition
F4:
![image](https://cloud.githubusercontent.com/assets/1945821/17122125/8d9e5a16-5274-11e6-9b37-4263e3295df5.png)

F446
![image](https://cloud.githubusercontent.com/assets/1945821/17122133/a4a69f34-5274-11e6-87fa-f963a1a5c756.png)

Note bit 21 is now an 1 is enabled and use to be 1: VBUS sensing not available by hardware

What is the correct way to specialize a version of a CPU in libopencm3?

**Comment by ChuckM** [comment]
It is a challenge. If you wanted to write software that worked on both your only choice would be to decide what to do based on CPUID (and not the standard one, the proprietary one) so code like this might work:

```
#define ST_CPUID             MMIO32(0xe0042000);

void
set_vbus() {
    if (ST_CPUID & 0xfff) == 0x434) ||  (ST_CPUID & 0xfff) == <446 id>) {
          OTG_GCCFG |= 1 << 21;
   }
}
```

There is no "variant specific" indicator in the build or the makefiles which you can use and the same library is going to be created for all F4 chips. Note that the ARM architecture manual's SCB for both processor gives the same part number 0xC24. So you can't use that to distinguish. The best of a bad situation is an API call in the USB stack which sets this bit and looks in the CPU's panties to see which kind it is before operating.
}

**Comment by nschodet-sevenhugs** [comment]
For USB, what about OTG_CID register? It is 0x2000 on f479/f469.

**Comment by karlp** [comment]
The USB issues were committed back in cf80e2bd5e19eadd8fe0e8719070fc59070a3459
The PLLR bits were fixed back in 57c2b00a69f97205313e1c7ab8116ee1893b231e  Sorry I didnt' close this ticket way back then.
